langlib/lang.value/src/main/ballerina/src/lang.value/value.bal

```diff
@@ -140,13 +140,31 @@ public function toJson(anydata v) returns json = external;
 public function toJsonString(anydata v) returns string = external;
 
 # Parses a string in JSON format and returns the the value that it represents.
-# All numbers in the JSON will be represented as float values.
+# All integer numbers in the JSON will be represented as integer values.
+# All decimal numbers except -0.0 in the JSON will be represent as decimal values.
+# -0.0 in the JSON will be represent as float value.
 # Returns an error if the string cannot be parsed.
 #
 # + str - string representation of json
 # + return - `str` parsed to json or error
 public function fromJsonString(string str) returns json|error = external;
 
+# Parses a string in JSON format and returns the value that it represents.
+# All numbers in the JSON will be represented as float values.
+# Returns an error if the string cannot be parsed.
+#
+# + str - string representation of json
+# + return - `str` parsed to json or error
+public function fromJsonFloatString(string str) returns json|error = external;
```

Comment:
Return type of `fromJsonFloatString` should be `JsonFloat`. This will fix with https://github.com/ballerina-platform/ballerina-lang/issues/25446